### PR TITLE
[#23] [#24] [UI] [Integrate] Dropdown answer

### LINKED ISSUE #23: [UI] As a user, I can see the dropdown question when taking a survey
## Why

App must display the single dropdown fetched from backend

## Acceptance Criteria

- App must display single dropdown fetched from backend answering the survey
- Dropdown must be a spinner that use can scroll through

## Resources

- [Design](https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=48%3A0&t=LtygnfHfEHpIc5VD-0)

### LINKED ISSUE #24: [Integrate] As a user, I can see the dropdown question when taking a survey
## Why

App must display the single dropdown fetched from backend

## Acceptance Criteria

- App must display single dropdown fetched from backend answering the survey
- Must display question with `displayType` "dropdown"

## Resources

- [Design](https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=48%3A0&t=LtygnfHfEHpIc5VD-0)

### PR DESCRIPTION
**Pull request by lydiasama**
- Closes https://github.com/nimblehq/flutter-ic-lydia-ryan/issues/23
- Closes https://github.com/nimblehq/flutter-ic-lydia-ryan/issues/24

## What happened 👀

Implemented UI and integration for dropdown answer.

## Insight 📝

- Used [scroll_snap_list: ^0.9.1](https://pub.dev/packages/scroll_snap_list) library to make it snap the focused text at the center.
- As I ran it on the simulator, I could not test the swipe gesture. By the way, this library support `focusOnItemTap` so we can select it by tapping on the item.

## Proof Of Work 📹

https://github.com/nimblehq/flutter-ic-lydia-ryan/assets/12026942/61df882b-e1db-459e-adfe-86fe46c6e6ec

### CODE REVIEW

File: lib/screens/surveydetails/widget/answer_dropdown_widget.dart [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+const double _dropdownThreshold = 20.0;
+
+class AnswerDropdownWidget extends StatefulWidget {
+  final QuestionUiModel question;
+  final Function(String) onSelected;
+
+  const AnswerDropdownWidget({
+    Key? key,
+    required this.question,
+    required this.onSelected,
```
Comment (markgravity) [inline lib/screens/surveydetails/widget/answer_dropdown_widget.dart]: What do you think about `onSelect`?

Comment (lydiasama) [reply lib/screens/surveydetails/widget/answer_dropdown_widget.dart]: @markgravity  Resolved in 5f651357 👍🏻

**Review by markgravity — APPROVED** [review]

**Comment by ryan-conway** [comment]
@lydiasama Mind resolving conflicts? 😄

**Comment by lydiasama** [comment]
@ryan-conway Because you merged the refactoring first. 😂  I resolved it. 🙌🏻 
FYI, Please merge #101   first. 🙏🏻

File: lib/screens/surveydetails/widget/answer_dropdown_widget.dart [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+              child: ScrollSnapList(
+                scrollDirection: Axis.vertical,
+                onItemFocus: _onItemFocus,
+                itemSize: Dimensions.answerDropdownHeight,
+                itemBuilder: _buildListItem,
+                itemCount: widget.question.answers.length,
+              ),
+            ),
+          ],
+        ),
```
Comment (manh-t) [inline lib/screens/surveydetails/widget/answer_dropdown_widget.dart]: @lydiasama do we need `Column` in this widget? It has only one child, so I think we can optimize to:
```suggestion
      child: SizedBox(
        height: (_dropdownItemLength * Dimensions.answerDropdownHeight) -
            _dropdownThreshold,
        child: ScrollSnapList(
                scrollDirection: Axis.vertical,
                onItemFocus: _onItemFocus,
                itemSize: Dimensions.answerDropdownHeight,
                itemBuilder: _buildListItem,
                itemCount: widget.question.answers.length,
              ),
```

Comment (lydiasama) [reply lib/screens/surveydetails/widget/answer_dropdown_widget.dart]: @manh-t I'm not sure if it affects to the `ScrollSnapList`. Let me try. 🙏🏻

Comment (lydiasama) [reply lib/screens/surveydetails/widget/answer_dropdown_widget.dart]: @manh-t It works! 👍🏻  Actually, I tried but it didn't work for some reason. Maybe after I refactored the UI could help. BTW, thank you for spotting on this.

Comment (lydiasama) [reply lib/screens/surveydetails/widget/answer_dropdown_widget.dart]: Fixed in 1406edd4

**Review by ngocmanh1609 — APPROVED** [review]
good job 👏

**Review by manh-t — APPROVED** [review]
good job 👏

**Pull request merged**